### PR TITLE
wait LRO

### PR DESCRIPTION
**Pull request by qiqingzhang**
public pr for wait LRO
linked to https://github.com/hashicorp/terraform-provider-azurerm/issues/15641

### CODE REVIEW

File: internal/services/automation/automation_runbook_resource.go [hunk 1]
```diff
@@ -201,10 +201,14 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
...
 		draftClient := meta.(*clients.Client).Automation.RunbookDraftClient
 
-		_, err := draftClient.ReplaceContent(ctx, id.ResourceGroup, id.AutomationAccountName, id.Name, reader)
+		future, err := draftClient.ReplaceContent(ctx, id.ResourceGroup, id.AutomationAccountName, id.Name, reader)
 		if err != nil {
 			return fmt.Errorf("setting the draft for %s: %+v", id, err)
 		}
+		if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("waiting for update of %q: %+v", id, err)
+		}
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_runbook_resource.go:210]: per below, this is still blocked on this issue? https://github.com/Azure/azure-sdk-for-go/issues/17196

**Review by katbyte — CHANGES_REQUESTED** [review]
looks like this is still blocked on the api/sdk
```
Error: waiting for update of "Runbook: (Name \"Get-AzureVMTutorial\" / Automation Account Name \"acctest-220815194529607510\" / Resource Group \"acctestRG-auto-220815194529607510\")": pollingTrackerPut#checkForErrors: the response did not contain a body: StatusCode=0
```

**Comment by ms-henglu** [comment]
Hi @katbyte , thanks! I've removed the changes and resolved the conflicts.

**Comment by katbyte** [comment]
It seems like the delete is returning early and not after everything has been removed?
```
------- Stdout: -------
=== RUN   TestAccTemplateDeployment_withParamsBody
=== PAUSE TestAccTemplateDeployment_withParamsBody
=== CONT  TestAccTemplateDeployment_withParamsBody
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Resource Group "acctestRG-220817182841153205": the Resource Group still contains Resources.
        
        Terraform is configured to check for Resources within the Resource Group when deleting the Resource Group - and
        raise an error if nested Resources still exist to avoid unintentionally deleting these Resources.
        
        Terraform has detected that the following Resources still exist within the Resource Group:
        
        * `/subscriptions/*******/resourceGroups/acctestRG-220817182841153205/providers/Microsoft.Network/publicIPAddresses/myPublicIpqh7cd4rsruqhq`
        * `/subscriptions/*******/resourceGroups/acctestRG-220817182841153205/providers/Microsoft.Storage/storageAccounts/qh7cd4rsruqhqstorage`
        
        This feature is intended to avoid the unintentional destruction of nested Resources provisioned through some
        other means (for example, an ARM Template Deployment) - as such you must either remove these Resources, or
        disable this behaviour using the feature flag `prevent_deletion_if_contains_resources` within the `features`
        block when configuring the Provider, for example:
        
        provider "azurerm" {
          features {
            resource_group {
              prevent_deletion_if_contains_resources = false
            }
          }
        }
        
        When that feature flag is set, Terraform will skip checking for any Resources within the Resource Group and
        delete this using the Azure API directly (which will clear up any nested resources).
        
        More information on the `features` block can be found in the documentation:
        https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#features
        
        
--- FAIL: TestAccTemplateDeployment_withParamsBody (939.84s)
```

File: internal/services/resource/template_deployment_resource_test.go [hunk 2]
```diff
@@ -408,7 +408,11 @@ DEPLOY
 func (TemplateDeploymentResource) withParamsBody(data acceptance.TestData) string {
 	return fmt.Sprintf(`
 provider "azurerm" {
-  features {}
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
```
Comment (katbyte) [inline internal/services/resource/template_deployment_resource_test.go:415]:
```suggestion
  features {}
```

**Review by katbyte — APPROVED** [review]
The failure i still there, but going to merge this because it is in main too. eveything else looks good

**Pull request merged**